support-lib/jni/djinni_support.cpp

```diff
@@ -22,6 +22,7 @@
 #include <cstring>
 
 static_assert(sizeof(jlong) >= sizeof(void*), "must be able to fit a void* into a jlong");
+static_assert(sizeof(wchar_t) == 2 || sizeof(wchar_t) == 4, "wchar_t must be represented by UTF-16 or UTF-32 encoding");
```

Comment:
Yeah, but I was assuming if you put it inside of template functions you'd assert on a type argument.  I guess what you're saying is that a static_assert(false) inside of the default implementation of the template functions below wouldn't work.  But a static_assert(wcharTypeSize == 2 || wcharTypeSize == 4) would work, right?  The checking is redundant, but the use of the template parameter will delay the evaluation.